src/tracks/InstrumentTrack.cpp

```diff
@@ -249,6 +263,25 @@ MidiEvent InstrumentTrack::applyMasterKey( const MidiEvent& event )
 
 
 
+void InstrumentTrack::processCCEvent(int controller)
+{
+	// Does nothing if the LED is disabled
+	if( !m_midiCCEnable->value() )
+	{
+		return;
+	}
+
+	uint8_t channel = static_cast<uint8_t>( midiPort()->realOutputChannel() );
```

Comment:
I missed this, but does it make sense to use the output channel with `processInEvent`?

Reply:
I think it does, because the event isn't actually being received by the input MIDI ports, but being generated by the track and then forwarded to the instrument plugin. The reason I used `processInEvent` was for it to be processed just as a MIDI event received through a physical controller would be (there are some CC events that are handled by LMMS like sustain events for example). Since the notes generated by the track are forwarded to the plugin through the output channel, I thought generated CC messages should be too.

You think it makes more sense to use the input channel instead?

Reply:
Hard to decide. Can we compare with other DAWs?

Reply:
Of course! If other DAWs use the input channel instead or if most people think it's a better alternative we can change it. Personally I thought using the output channel would be more intuitive. Whichever we decide for, we should document it on the LMMS wikia since it's something the regular user can't easily find out (since with `processInEvent` processing the event we can't use a MIDI monitor to see the channel being used afaik, so the user has to know it before hand).